lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -711,6 +725,25 @@ public void run() {
 		}
 	}
 
+	private void handleReauth() {
+		pendingReauth = false;
```

Comment:
Should this only be cleared once reauth is successful? If we don't clear it here, and the `authCallback` fails (with an exception), when would there be a new attempt? The next time the thread exits `tryWait()` ?